### PR TITLE
Implement no-op HTTP/2 IHttpUpgradeFeature

### PR DESCRIPTION
**Pull request by halter73**
Addresses #3081 

@muratg We'd like to get this in for 2.2

### CODE REVIEW

File: test/Kestrel.Core.Tests/HttpProtocolFeatureCollectionTests.cs [hunk 1]
```diff
@@ -19,22 +19,30 @@ namespace Microsoft.AspNetCore.Server.Kestrel.Core.Tests
...
                 ConnectionFeatures = new FeatureCollection(),
                 TimeoutControl = Mock.Of<ITimeoutControl>(),
                 Transport = Mock.Of<IDuplexPipe>(),
+                ServerPeerSettings = new Http2PeerSettings(),
+                ClientPeerSettings = new Http2PeerSettings(),
             };
 
-            _http1Connection = new TestHttp1Connection(_http1ConnectionContext);
+            _httpConnectionContext = context;
+            _http1Connection = new TestHttp1Connection(context);
```
Comment (JunTaoLuo) [inline test/Kestrel.Core.Tests/HttpProtocolFeatureCollectionTests.cs]: Http1Connection with a Http2StreamContext?

File: test/Kestrel.Core.Tests/HttpProtocolFeatureCollectionTests.cs [hunk 2]
```diff
@@ -143,24 +151,22 @@ public void FeaturesSetByGenericSameAsByType()
...
-            Assert.Null(http2StreamCollection.Get<IHttpMinResponseDataRateFeature>());
+            Assert.Null(_http2Collection.Get<IHttpMinRequestBodyDataRateFeature>());
+            Assert.Null(_http2Collection.Get<IHttpMinResponseDataRateFeature>());
 
             Assert.NotNull(_collection.Get<IHttpMinRequestBodyDataRateFeature>());
             Assert.NotNull(_collection.Get<IHttpMinResponseDataRateFeature>());
         }
 
+        [Fact]
+        public void Http2StreamFeatureCollectionDoesIncludeUpgradeFeature()
```
Comment (JunTaoLuo) [inline test/Kestrel.Core.Tests/HttpProtocolFeatureCollectionTests.cs]: This is moved to HttpProtocol so is there an equivalent test to ensure it exist in Http1Connection?

File: test/Kestrel.Core.Tests/HttpProtocolFeatureCollectionTests.cs (hunk 1 above)
Comment (halter73) [reply test/Kestrel.Core.Tests/HttpProtocolFeatureCollectionTests.cs]: Http2StreamContext derives from HttpConnectionContext and these tests need just enough properties set on the context to not null-ref in the constructor.

File: test/Kestrel.Core.Tests/HttpProtocolFeatureCollectionTests.cs (hunk 2 above)
Comment (halter73) [reply test/Kestrel.Core.Tests/HttpProtocolFeatureCollectionTests.cs]: For Http1Connection, we have tests that actually use the feature.

**Comment by Eilon** [comment]
Approved for 2.2.

**Pull request merged**